acstools/focus_diverse_epsfs.py

```diff
@@ -0,0 +1,349 @@
+    simultaneously.
+
+    Returns
+    -------
+
+    Downloads the ePSF FITS files to download_location. The explicit return is 
+    a list of the downloaded FITS files.
+
+    """
+
+    # select if either choosing to provide inputs via a text file or list
+    if fromTextFile == True:
+        with open(input_list) as file:
+            lines = [line.rstrip() for line in file]
+
+    if fromTextFile == False:
+        lines = input_list
+
+    # perform multiprocessing with dask
+    download_locations = [download_location] * len(lines)
+    zipped_inputs = list(zip(lines, download_locations))
+
+    dask_results = [dask.delayed(psf_retriever)(
+        zipped_inputs[0], zipped_inputs[1]) for zipped_inputs in zipped_inputs]
+
+    # run with progress bar
+    with ProgressBar():
+        results = dask.compute(
+            *dask_results, num_workers=n_PROC, scheduler='processes')
+
+    return results
+
+
+def interp_epsf(ePSFs, x, y, pixel_space=False, subpixel_x=0, subpixel_y=0):
+    """
+    Function to perform further spatial interpolations given the input ePSF array from 
+    the previous retriever functions. The routine uses bi-linear interpolation for the 
+    integer pixel shifts, and bi-cubic interpolation for any specified subpixel phase shifts.
+
+    Returns
+    -------
+    An ePSF array with the specified interpolation parameters.  
+
+    """
+
+    # input error checking
+    if x not in range(0, 4096):
+        msg = ("The x coordinate should be an integer between 0 and 4096. \n \
+            Please double-check your inputs.")
+        LOG.error(msg)
+        return
+
+    if y not in range(0, 4096):
+        msg = ("The y coordinate should be an integer between 0 and 4096. \n \
+            Please double-check your inputs.")
+        LOG.error(msg)
+        return
+
+    if subpixel_x < 0.0 or subpixel_x > 0.99:
+        msg = ("Subpixel shifts should be between 0 and 0.99. \n \
+            Please double-check your inputs.")
+        LOG.error(msg)
+        return
+
+    if subpixel_y < 0.0 or subpixel_y > 0.99:
+        msg = ("Subpixel shifts should be between 0 and 0.99. \n \
+            Please double-check your inputs.")
+        LOG.error(msg)
+        return
+
+    if pixel_space not in [True, False]:
+        msg = ("pixel_space should be True or False.\n \
+            Please double-check your inputs.")
+        LOG.error(msg)
+        return
+
+    if pixel_space == False and (subpixel_x != 0 or subpixel_y != 0):
+        msg = ('Please set pixel_space = True to use the subpixel phase offsets.')
+        LOG.error(msg)
+        return
+
+    # round subpixel_x and subpixel_y to second decimal
+    subpixel_x = round(subpixel_x, 2)
+    subpixel_y = round(subpixel_y, 2)
+
+    # if user inputs coordinates for WFC1 (y > 2048), then adjust down and set chip = "WFC1",
+    # otherwise assume chip = "WFC2"
+    chip = "WFC2"
+
+    if y > 2048:
+        y = y-2048
+        chip = "WFC1"
+
```

Comment:
Make it very clear in the docstring/docs that the expected coordinates here are for the whole 4096x4096 frame, not per chip. I'm not 100% sure I agree with this choice, since I'd imagine most people will be choosing their desired PSF location in an FLT/FLC, which has separate arrays for each chip. Maybe there's a way to enable both full-frame and per chip coordinates?